one_fm/utils.py

```diff
@@ -3640,4 +3640,45 @@ def send_work_anniversary_reminders():
                     others = [d for d in anniversary_persons if d != person]
                     reminder_text = get_work_anniversary_reminder_text(others)
                     send_work_anniversary_reminder(person_email, reminder_text, others, message, sender)
-                    
\ No newline at end of file
+
+
+def set_employee_status_to_vacation():
+    # Get today's date
+    current_date = today()
+
+    # Fetch all approved leave applications where `from_date` is today or earlier
+    leave_applications = frappe.get_all('Leave Application', 
+        filters={
+            'status': 'Approved',
+            'from_date': ['<=', current_date],
+            'to_date': ['>', current_date]
+        }, 
+        fields=['employee', 'from_date']
+    )
+
+    # Get a list of employees with approved leave applications
+    employees_on_leave = [leave['employee'] for leave in leave_applications]
+
+    if not employees_on_leave:
+        frappe.log_error(_("No employees found with approved leave applications for today or earlier."))
+        return
+
+    # Fetch all employees with 'Active' status
+    active_employees = frappe.get_all('Employee', 
+        filters={
+            'status': 'Active'
+        }, 
+        fields=['name']
+    )
+
+    # Filter active employees who are in the leave applications list
+    employees_to_set_vacation = [
```

Comment:
so this would not be necessary anymore

Reply:
DOne

Reply:
also, you still need to filter out inactive employees, 

so you need to get a list of name(id) of employee on leave, then also those that are active,

Reply:
I don't understand this. Why do I need filter out inactive

Reply:
@yusuffAFEX , so it is possible to have  employees with a different status from active, and the aim of this method is to set those with active status, to vacation, and let's assume, an employee is already set to vacation,  there would be no need to  set it again

Reply:
Yes, that's exactly what the list comprehension is doing. It is filter those who that have active status to set them to vacation.

Reply:
so, here, it is leave status that is being checked, not the employee status
![image](https://github.com/user-attachments/assets/b12afd01-cdbe-481b-bb6a-0338f5278503)

Reply:
It is employee status. I tested it

Reply:
Oh, great great, thank you very much